### DISCUSSION TITLE
How to disable window resize after the release of WinUI 3 1.0 Preview 3?

### DISCUSSION THREAD

**Discussion by JohnCido** [microsoft/WindowsAppSDK#1694] [Q&A] [answered]
After upgrading my project to Preview 3, I can see that AppWindowConfiguration was removed, so how should I disable window resizing now? Do I have to go back to the old DllImport methods again?

**Comment by martibravo** [comment, marked as answer]
This is quite easy to do with the new AppWindow APIs:

Add the following items to your MainWindow class:

```
using Microsoft.UI.Windowing;
using Microsoft.UI.Xaml.Interop;
using System.Windows.Interop.WindowsRuntime;
```
```
private AppWindow _apw;
private OverlappedPresenter _presenter;

public void GetAppWindowAndPresenter() {
	var hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);
	WindowId myWndId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(hWnd);
	_apw = AppWindow.GetFromWindowId(myWndId);
	_presenter = _apw.Presenter as OverlappedPresenter;
}
```

and in the MainWindow contructor:

```
GetAppWindow();
_presenter.IsResizable = false;
```

**Reply by nickrandolph** [reply]
@martibravo thanks for this explanation. Can I ask why it makes sense for the IsResizable property to be on the presenter, when most of the other window APIs are now directly on the AppWindow?

**Reply by martibravo** [reply]
> @martibravo thanks for this explanation. Can I ask why it makes sense for the IsResizable property to be on the presenter, when most of the other window APIs are now directly on the AppWindow?

If you check the docs, you'll see that the AppWindow class properties are related to Window management (is it shown in the Taskbar? Move it here... maximize it!) and the Presenter properties and methods are related to How the window looks: should it have borders? Should it have a titlebar? Should it be resizable? Should it be fullscreen?

**Reply by nickrandolph** [reply]
@martibravo thanks for clarifying that - looking through the rest of the APIs on the AppWindow, AppWindowTitleBar and the presenters, I can see the logical separation of features now. I was just surprised as there was quite a big change between the previews

Not sure which docs you're referring to - the only page I can find is this (https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/windowing/windowing-overview) which provides a very superficial overview. Was there a full spec prepared for this? Even the .NET projections don't seem to have xml comments, making it hard for developers to explore the API surface.

**Reply by martibravo** [reply]
@nickrandolph Yeah I was referring to those docs and the API reference on that same webpage—there's a litte description of every property, method and event in the Windowing namespace. But I agree, docs should be better. Also, have you looked at the [WindowsAppSDK windowing samples](https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/Windowing)? That's where I learned how this API actually works.

**Reply by nickrandolph** [reply]
Yeh I checked them out. The samples are pretty comprehensive, which is fantastic.

**Reply by rcohn** [reply]
Hi @martibravo -

Thank you for the very clear code example and refs above - it nicely works to keep my WinUI3 app from being resizable.  I was not aware of the Presenter API (first I'd seen it being used in the many code examples and docs I've been looking through).  I hope there is a Presenter API that lets me set the AppWindow size as well.

Again, my thanks.
